### DISCUSSION TITLE
Resyncing Commands

### DISCUSSION THREAD

**Discussion by cerqiest** [Pycord-Development/pycord#1683] [Q&A] [answered]
Hi there. This is my first post here so bear with me.
I'm trying to make a (slash) command that reloads all the bot's cogs.
I have successfully done this, however I have one problem.
I need to know how I can resync all the commands on reload without restarting my entire bot.
For example, if I had a bot running and added a command, it wouldn't register with Discord.
PyCord doesn't seem to do this automatically.

I've tried using the sync_commands method and that just threw an error that command names must be unique.
Thank you!

**Comment by JustaSqu1d** [comment, marked as answer]
`sync_commands` is only called to sync un-synced commands, ones that are not on Discord. You should only sync the commands that have been changed.

Discussions here are usually development-related. Please use [#help](https://discord.gg/bWYRHqRRWu) instead.
